test/shared_utils/compare.py

```diff
@@ -5,13 +5,68 @@
 This module focuses on comparing output specifically with xarray.
 """
 
+import json
 from itertools import count
 
-
 from xarray.backends.api import open_groups
 from xarray.core.datatree import DataTree
 from xarray import Dataset
 
+from reference_hashing import get_hashes_from_xarray_input, XARRAY_DECODE_DEFAULTS
+
+
+def matches_reference_hash_file_using_xarray(
+    request_output_path: str,
+    reference_file_path: str,
+    skipped_variables_or_groups: set[str] = set(),
+    skipped_metadata_attributes: set[str] = set(),
+    xarray_kwargs: dict = XARRAY_DECODE_DEFAULTS,
+) -> bool:
+    """Generate hashes for request output and compare to reference file.
+
+    Args:
+        request_output_path: netCDF4 or HDF5 file retrieved from a Harmony
+        request.
+        reference_file_path: File containing generated SHA256 values for every
+        group and variable in the original test output.
+        skipped_variables_or_groups: Variables or groups that are known to vary
+        between different test executions. For example, `/subset_files` in the
+        output from SAMBAH, which varies between production and UAT.
+        skipped_metadata_attributes: Names of metadata attributes to omit from
+        the derivation of the SHA256 hash for all group and variable metadata.
+        These will be values that are known to vary and are in addition to
+        `history` and `history_json`. The main use-case is metadata attributes
+        with timestamps dependent on request execution time.
+        xarray_kwargs: dict containing arguments used by `xarray` to open the
+        request output file as a `DataTree` object. Default is to switch off all
```

Comment:
This could be formatted differently
```suggestion
          Args:
        request_output_path: netCDF4 or HDF5 file retrieved from a Harmony
             request.
        reference_file_path: File containing generated SHA256 values for every
             group and variable in the original test output.
             skipped_variables_or_groups: Variables or groups that are known to
             vary between different test executions. For example,
             `/subset_files` in the output from SAMBAH, which varies between
             production and UAT.
        skipped_metadata_attributes: Names of metadata attributes to omit from
             the derivation of the SHA256 hash for all group and variable
             metadata.  These will be values that are known to vary and are in
             addition to `history` and `history_json`. The main use-case is
             metadata attributes with timestamps dependent on request execution
             time.
        xarray_kwargs: dict containing arguments used by `xarray` to open the
             request output file as a `DataTree` object. Default is to switch
             off all decoding options.

```